File: sdk/ruby/test/fixtures/run_worker.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

# Entry point launched by Portals as the direct worker process (the socket
# path is appended as the final CLI argument by `Portals.Connection`). Puts
# this fixtures directory and the SDK's `lib` on the load path so both
# `bench_worker` and `portals` are requirable, then runs the worker loop.

here = File.expand_path(__dir__)
$LOAD_PATH.unshift(here)
$LOAD_PATH.unshift(File.expand_path('../../lib', here))

require 'portals'
require 'bench_worker'
require 'conformance_worker'

Portals::Worker.new(
  max_concurrency: Integer(ENV.fetch('PORTALS_MAX_CONCURRENCY', '64')),
  max_streams: Integer(ENV.fetch('PORTALS_MAX_STREAMS', '8')),
  protocol_version: Integer(ENV.fetch('PORTALS_PROTOCOL_VERSION', Portals::Protocol::PROTOCOL_VERSION.to_s)),
  transport: ENV.fetch('PORTALS_TRANSPORT', 'unix').to_sym
).run

File: sdk/node/test/fixtures/run_worker.js
#!/usr/bin/env node
'use strict';

// Entry point launched by Portals as the direct worker process (the socket
// path is appended as the final CLI argument by `Portals.Connection`).
// Registers the fixture modules under the names the BEAM uses in `CALL`,
// then runs the worker loop.

const { Worker } = require('../..');

const worker = new Worker({
  maxConcurrency: Number(process.env.PORTALS_MAX_CONCURRENCY ?? 64),
  maxStreams: Number(process.env.PORTALS_MAX_STREAMS ?? 8),
  protocolVersion: Number(process.env.PORTALS_PROTOCOL_VERSION ?? 1),
  transport: process.env.PORTALS_TRANSPORT ?? 'unix',
  modules: {
    bench_worker: require('./bench_worker'),
    conformance_worker: require('./conformance_worker'),
  },
  modulePaths: [__dirname],
});

worker.run().catch((err) => {
  process.stderr.write(`portals worker: ${err.stack || err}\n`);
  process.exit(1);
});
